File: web/src/parse/parseBlock.ts
import type BlockParserState from "../types/BlockParserState";
import type MarkdownNode from "../types/MarkdownNode";
import isEscaped from "../utils/isEscaped";

export default function parseBlock(
	state: BlockParserState,
	parent: MarkdownNode,
	endOfLine: number,
): void {
	state.isEscaped = isEscaped(state.src, state.i);

	for (let rule of state.rules) {
		//let start = state.i;
		let handled = rule.testStart(state, parent, endOfLine);

		if (handled) {
			//if (state.debug) {
			//	console.log(`Found ${rule.name}, at ${start}`);
			//}

			// DEBUG: Make sure we are AFTER the line end?

			return;
		}
	}
}

File: dotnet/Allmark/Parse/ParseBlock.cs
namespace Allmark.Parse;

using Allmark.Types;

public static class ParseBlock
{
    public static void Execute(BlockParserState state, MarkdownNode parent)
    {
        state.IsEscaped = Utils.IsEscaped(state.Src, state.I);

        foreach (var rule in state.Rules)
        {
            // int start = state.I;
            bool handled = rule.TestStart(state, parent);

            if (handled)
            {
                // if (state.Debug)
                // {
                // 	Console.WriteLine($"Found {rule.Name}, at {start}");
                // }

                // DEBUG: Make sure we are AFTER the line end?

                return;
            }
        }
    }
}
